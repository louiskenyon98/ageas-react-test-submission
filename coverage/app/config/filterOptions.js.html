<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/config/filterOptions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">app/config</a> filterOptions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/3</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const <span class="cstat-no" title="statement not covered" >LAST_CREATED = "createdAt";</span>
export const <span class="cstat-no" title="statement not covered" >LAST_EDITED = "updatedAt";</span>
export const <span class="cstat-no" title="statement not covered" >filterOptions = [{</span>
    value: LAST_CREATED,
    label: 'Last created'
},{
    value: LAST_EDITED,
    label: 'Last edited'
}];
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Apr 23 2019 09:55:11 GMT+0100 (GMT Daylight Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
