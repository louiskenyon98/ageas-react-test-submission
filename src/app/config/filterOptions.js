export const LAST_CREATED = "createdAt";
export const LAST_EDITED = "updatedAt";
export const filterOptions = [{
    value: LAST_CREATED,
    label: 'Last created'
},{
    value: LAST_EDITED,
    label: 'Last edited'
}];
