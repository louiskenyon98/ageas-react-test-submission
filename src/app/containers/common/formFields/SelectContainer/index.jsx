import React from 'react';
import {connect} from 'react-redux';
import Select from '../../../../components/common/formFields/Select';
import {filterOptions} from '../../../../config/filterOptions';

import {onChangeFilterValue} from '../../../../actions/filterValues/index.actions';


class SelectContainer extends React.Component {
    render() {
        return (
            <Select
                options={filterOptions}
                value={this.props.filterOption}
                onChange={this.props.onChangeFilterValue}
            />
        )
    }
}

const mapStateToProps = (state) => {
    return {
        filterOption: state.notes.filterOption
    }
};
export default connect(mapStateToProps, {onChangeFilterValue})(SelectContainer);
