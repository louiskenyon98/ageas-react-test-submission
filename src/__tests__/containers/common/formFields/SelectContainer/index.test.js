import React from 'react';
import {shallow} from 'enzyme';
import {LAST_CREATED} from '../../../../../app/config/filterOptions';
import {SelectContainer} from '../../../../../app/containers/common/formFields/SelectContainer/index';

//:todo Get Mike to check this one.

describe('SelectContainer', () => {
    let props = {};
    let wrapper = () => shallow(<SelectContainer {...props}/>);
    beforeEach(() => {
        props = {
            filterOption: LAST_CREATED,
            onChangeFilterValue: jest.fn()
        }
    });
    describe('render', () => {
        it('should render a select component with options', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
    describe('functionality', () => {
        it('should call onChangeFilterValue onChange', () => {
            expect(props.onChangeFilterValue).not.toHaveBeenCalled();
            wrapper().find('Select').first().props().onChange('value');
            expect(props.onChangeFilterValue).toHaveBeenCalledTimes(1);
            expect(props.onChangeFilterValue).toHaveBeenCalledWith('value');

        })
    })
});
