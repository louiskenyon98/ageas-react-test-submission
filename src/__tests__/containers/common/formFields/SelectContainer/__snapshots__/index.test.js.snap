// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectContainer render should render a select component with options 1`] = `
<Select
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "label": "Last created",
        "value": "createdAt",
      },
      Object {
        "label": "Last edited",
        "value": "updatedAt",
      },
    ]
  }
  value="createdAt"
/>
`;
